WalletWasabi.Tests/UnitTests/MempoolMirrorTests/MirrorMempoolTests.cs

```diff
@@ -0,0 +1,185 @@
+			try
+			{
+				var rpc = coreNode.RpcClient;
+				var network = rpc.Network;
+				var mempoolInstance = services.Get<MempoolMirror>();
+				var walletName = "RandomWalletName";
+				await rpc.CreateWalletAsync(walletName);
+
+				var spendAmount = new Money(0.0004m, MoneyUnit.BTC);
+				var spendAmount2 = new Money(0.0004m, MoneyUnit.BTC);
+				await rpc.GenerateAsync(101);
+
+				var txid = await rpc.SendToAddressAsync(BitcoinFactory.CreateBitcoinAddress(network), spendAmount);
+				var txid2 = await rpc.SendToAddressAsync(BitcoinFactory.CreateBitcoinAddress(network), spendAmount2);
+
+				await services.StartAllAsync();
+				await mempoolInstance.TriggerAndWaitRoundAsync(TimeSpan.FromSeconds(7));
+				var localMempoolHashes = mempoolInstance.GetMempoolHashes();
+
+				Assert.Equal(2, localMempoolHashes.Count);
+				Assert.Contains(txid, localMempoolHashes);
+				Assert.Contains(txid2, localMempoolHashes);
+			}
+			finally
+			{
+				await services.StopAllAsync();
+				await coreNode.TryStopAsync();
+			}
+		}
+
+		[Fact]
+		public async Task CanHandleArrivingTxAsync()
+		{
+			var coreNode = await TestNodeBuilder.CreateAsync();
+			using HostedServices services = new();
+			services.Register<MempoolMirror>(new MempoolMirror(TimeSpan.FromSeconds(7), coreNode.RpcClient, coreNode.P2pNode), "Mempool Mirror");
+			try
+			{
+				var rpc = coreNode.RpcClient;
+				var network = rpc.Network;
+				var walletName = "RandomWalletName";
+				await rpc.CreateWalletAsync(walletName);
+
+				await services.StartAllAsync();
+
+				var spendAmount = new Money(0.0004m, MoneyUnit.BTC);
+				await rpc.GenerateAsync(101);
+
+				var txid = await rpc.SendToAddressAsync(BitcoinFactory.CreateBitcoinAddress(network), spendAmount);
+
+				Thread.Sleep(3000);
+
+				var mempoolInstance = services.Get<MempoolMirror>();
+				await mempoolInstance.TriggerAndWaitRoundAsync(TimeSpan.FromSeconds(21));
+				var localMempoolHashes = mempoolInstance.GetMempoolHashes();
+
+				Assert.Single(localMempoolHashes);
+				Assert.Contains(txid, localMempoolHashes);
+			}
+			finally
+			{
+				await services.StopAllAsync();
+				await coreNode.TryStopAsync();
+			}
+		}
+
+		[Fact]
+		public async Task CanHandleTheSameTxSentManyTimesAsync()
+		{
+			var coreNode = await TestNodeBuilder.CreateAsync();
+			using HostedServices services = new();
+			services.Register<MempoolMirror>(new MempoolMirror(TimeSpan.FromSeconds(2), coreNode.RpcClient, coreNode.P2pNode), "Mempool Mirror");
+			try
+			{
+				var rpc = coreNode.RpcClient;
+				var network = rpc.Network;
+				await services.StartAllAsync();
+				var mempoolInstance = services.Get<MempoolMirror>();
+
+				var walletName = "RandomWalletName";
+				await rpc.CreateWalletAsync(walletName);
+
+				using var k1 = new Key();
+				var blockId = await rpc.GenerateToAddressAsync(1, k1.PubKey.WitHash.GetAddress(network));
+				var block = await rpc.GetBlockAsync(blockId[0]);
```

Comment:
Even tough in this test case `blockId` contains one block's id, it's an array, so it should be named `blockIds`.

Reply:
done